newIDE/app/src/ObjectsList/index.js

```diff
@@ -418,205 +609,235 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
 
         const newObjectWithContext = addSerializedObjectToObjectsContainer({
           objectName: name,
-          positionObjectName: pasteObject.getName(),
+          positionObjectFolderOrObjectWithContext: objectFolderOrObjectWithContext,
           objectType: type,
           serializedObject: copiedObject,
-          global,
+          addInsideFolder,
         });
 
         onObjectModified(false);
         if (onObjectPasted) onObjectPasted(newObjectWithContext.object);
-
-        return newObjectWithContext;
+        if (addInsideFolder && treeViewRef.current)
+          treeViewRef.current.openItems([
+            getTreeViewItemId(objectFolderOrObjectWithContext),
+          ]);
       },
       [addSerializedObjectToObjectsContainer, onObjectModified, onObjectPasted]
     );
 
     const editName = React.useCallback(
-      (objectWithContext: ?ObjectWithContext) => {
-        onRenameObjectStart(objectWithContext);
-        // TODO Should it be called later?
-        if (sortableList.current) sortableList.current.forceUpdateGrid();
+      (objectFolderOrObjectWithContext: ?ObjectFolderOrObjectWithContext) => {
+        if (!objectFolderOrObjectWithContext) return;
+        if (treeViewRef.current)
+          treeViewRef.current.renameItem(objectFolderOrObjectWithContext);
       },
-      [onRenameObjectStart]
+      []
     );
 
     const duplicateObject = React.useCallback(
-      (objectWithContext: ObjectWithContext) => {
-        const { object, global } = objectWithContext;
+      (
+        objectFolderOrObjectWithContext: ObjectFolderOrObjectWithContext,
+        duplicateInScene?: boolean
+      ) => {
+        const {
+          objectFolderOrObject,
+          global,
+        } = objectFolderOrObjectWithContext;
+        if (objectFolderOrObject.isFolder()) return;
 
+        const object = objectFolderOrObject.getObject();
         const type = object.getType();
         const name = object.getName();
         const serializedObject = serializeToJSObject(object);
 
         const newObjectWithContext = addSerializedObjectToObjectsContainer({
           objectName: name,
-          positionObjectName: name,
+          positionObjectFolderOrObjectWithContext: objectFolderOrObjectWithContext,
           objectType: type,
           serializedObject,
-          global,
         });
 
-        editName(newObjectWithContext);
+        const newObjectFolderOrObjectWithContext = {
```

Comment:
Question: from your experience working on this @AlexandreSi, do you think we're far from being able to remove entirely this "WithContext" intermediate object. I.e: where is `global` used and what do we need to entirely remove our reliance on it?

(this is not something to address in this PR)

Reply:
(it's possible that we can't remove it, but one day we will rework `global` so that instead of being true/false, it will be a reference to the ObjectsContainer - so we can work with any number of "scope" of objects, in case we have folder of scenes)

Reply:
Well, `global` is only used to know which container operations should be done on. I haven't really thought about the idea of removing this but it could be more convenient indeed